src/types.rs

```diff
@@ -137,3 +148,99 @@ fn field_name_to_sql_with_alias() {
         "\"users\" AS new_users".to_string()
     );
 }
+
+// NOTE: GQLArgs is a simplified version of the
+// AST's representation of arguments. Using
+// such a structure only because it's easier
+// to do operations. The AST is much more
+// sophisticated and hence can become cumbersome
+#[derive(Serialize, Clone, Debug)]
+pub enum GQLArgType {
+    // NOTE: supported for [distinct_on]
+    String(String),
+    // NOTE: supported for [limit, offset]
+    Int(i64),
+    // Float(f32)
+}
+
+type FieldArguments = indexmap::IndexMap<String, GQLArgType>;
+
+impl GQLArgType {
+    // NOTE: functions like the one's defined below
+    // should be limited in usage since we can go
+    // wrong quite easily since we're relying on the
+    // developer's conscience over the type system
+    // hence opening up an avenue for bugs and issues
+
+    pub fn get_num(&self) -> i64 {
+        if let GQLArgType::Int(num) = self {
+            return *num;
+        }
+        // FIXME?: This should/would never happen
+        // unless we use it incorrectly
+        0
+    }
+
+    pub fn get_string(&self) -> String {
+        if let GQLArgType::String(txt) = &self {
+            return txt.to_string();
+        }
+        // FIXME?: This should/would never happen
+        // unless we use it incorrectly
+        "".to_string()
+    }
+}
+
+#[derive(Serialize, Clone, Debug)]
+pub struct FieldInfo {
+    pub fields: Vec<FieldName>,
+    pub root_field_arguments: indexmap::IndexMap<String, GQLArgType>,
+}
+
+impl FieldInfo {
+    pub fn new(fields: Vec<FieldName>, args: FieldArguments) -> FieldInfo {
+        FieldInfo {
+            fields,
+            root_field_arguments: args,
+        }
+    }
+}
+
+pub fn to_string_arg<'a>(
+    arg_name: String,
+    arg_val: &graphql_parser::query::Value<'a, &'a str>,
+) -> Result<(String, GQLArgType), error::GQLRSError> {
+    if let graphql_parser::query::Value::Enum(st) = arg_val {
+        return Ok((arg_name, GQLArgType::String(st.to_string())));
+    }
+
+    Err(error::GQLRSError::new(error::GQLRSErrorType::GenericError(
+        "failed to parse string".to_string(),
```

Comment:
Better error message
```suggestion
        format!("failed to parse argument {}", arg_name),
```